app/navigation/Navigator.scala

```diff
@@ -245,6 +245,19 @@ class Navigator @Inject() (categorisationService: CategorisationService) {
 
   private def navigateFromCategorisationPreparationPage(answers: UserAnswers, recordId: String): Call =
     answers.get(CategorisationDetailsQuery(recordId)) match {
+      case Some(catInfo)
```

Comment:
If you make the changes suggested to category info, then none of this new logic should be needed I think

Basically the navigator doesn't really need to know whether its niphls, if we set up the "categoryAssessmentsThatNeedAnswers" properly :)

Reply:
That make sense

Reply:
This now removed. But we kept the unit tests related NIPHLs